src/js/lib/logging/README.md

```diff
@@ -0,0 +1,3 @@
+A simple http server that takes POST requests for path "/" on localhost and writes their body to a new file in the "log" subdirectory of the current working directory
+
+To start the server, call `npm run start [-p <port>]` where `<port>` stands for the port to listen on. The default port is 20060.
```

Comment:
Why do you need this port number in `content.js`? Help info is provided by `client.js` and also content server is run from there? Regarding the log server - that's a very valid point. In such a case I would store it in some config, like `src/js/config.js` which will be imported by every package which needs it. (of course, add good docs to config.js and the variables stored there)

Reply:
The `client.js` currently passes the logging-server address to the window as a URL parameter. If you open the GUI by hand in an external web browser and don't submit this parameter then we need to get the address from somewhere else. This is why `content.js` also needs access to the default.

Reply:
The term config (or rather `cfg`) is already used in different places for different things, like `urlCfg`, `serverCfg`, `windowCfg` or just `cfg`. If it seems likely that we will add more to this one then i would refer to it as just `cfg` and rename the existing `cfg` in `content/index.js` to `urlCfg` which would be consistent with `client/index.js`. Is that fine or do you prefer another solution?

Reply:
Hmm, few thoughts / questions here:
1. There are 2 other options to handle it - what do you think about it?
    1. Make `client.js` always pass the default parameter even if its not explicitly provided in the url.
    2. In case the URL does not contain the option, just disable log sending.

2. Regarding naming - I like it, lets do it.

Reply:
> Make `client.js` always pass the default parameter even if its not explicitly provided in the url.

This seems like a good solution. The best way would likely be to pass the address through a cookie then.

> In case the URL does not contain the option, just disable log sending.

This would be trivial to implement. But is it intended that users will also open the IDE in their browser? If so, then this does not seem optimal to me.

There would also be the possible solution to send logs to the server that hosts the IDE. That server could then pass it on if necessary. But I don't know how the IDE will be used when it is finished, so I don't know what solution would  be best.

Reply:
> This seems like a good solution. The best way would likely be to pass the address through a cookie then.

Why it would be better than passing it as we pass it now, in url? TBH I'm not a web-guy, so I cant really compare these two solutions in my head now, so I'd love to know why cookies can be better here. The only thing that makes me instantly dislike cookies is that we would then need to display info in the app "that it uses cookies", which I would rather not want to display to the user :) 

> But is it intended that users will also open the IDE in their browser.

Opening in the browser was always meant to be a very debug thing and only for advanced users. Of course, I'm not talking about Enso Cloud, that we will host, but in this case we will pass the values in a special way (Im not sure how yet). On the other hand, the more I think about it, the more Im sure that you are right, that the browser experience should be close to electron experience as much as possible, so yep, lets not disable it then.

So yeah, I think the best way to progress with that is to use the common config js file that all needed scripts will use to refer to default values.

Reply:
The problem with passing it through the URL would only be that you need to enter it when you open the front end in your browser (if you don't use the default). Cookies are just one way for the IDE server to send the configuration in the background. They seem like the simplest solution given the implementation of your server.

But in my opinion, the current solution is fine for the moment and I would only change it when you know for sure that you need something different.